Docs: Clarify how to enable "Forward OAuth Identity" for data sources (#37696)
grafana/grafana: docs/sources/developers/plugins/add-authentication-for-data-source-plugins.md

If your data source uses the same OAuth provider as Grafana itself, for example using [Generic OAuth Authentication]({{< relref "../../auth/generic-oauth.md" >}}), your data source plugin can reuse the access token for the logged-in Grafana user.

> To allow Grafana to pass the access token to the plugin, the `jsonData.oauthPassThru` must be set to `true` on the data source configuration. The [DataSourceHttpSettings](https://developers.grafana.com/ui/latest/index.html?path=/story/data-source-datasourcehttpsettings--basic) provides a toggle for this as **Forward OAuth Identity**. Otherwise, you should build an appropriate toggle to set `jsonData.oauthPassThru` in your own data source configuration page UI.

Reviewer: ```suggestion
To allow Grafana to pass the access token to the plugin, update the data source configuration and set the` jsonData.oauthPassThru` property to `true`. The [DataSourceHttpSettings](https://developers.grafana.com/ui/latest/index.html?path=/story/data-source-datasourcehttpsettings--basic) provides a toggle, the **Forward OAuth Identity** option, for this. You can also build an appropriate toggle to set `jsonData.oauthPassThru` in your own data source configuration page UI.
```
